test/specs/test-documentation.el

```diff
@@ -40,12 +41,15 @@
         (let ((match (intern (match-string 1))))
           (cl-pushnew match matches))
         (-when-let (match (match-string 2))
+          (cl-pushnew (intern match) matches))
+        (-when-let (match (match-string 3))
           (cl-pushnew (intern match) matches))))
     matches))
 
 (defconst flycheck/checker-re
   (rx bol "   .. syntax-checker:: " (group (1+ nonl)) "\n"
-      (? "                       " (group (1+ nonl)) eol)))
+      (? "                       " (group (1+ nonl)) "\n")
+      (? "                       " (group (1+ nonl))) eol))
```

Comment:
As an alternative, would replacing the original `?` with a `*` work?

Reply:
The regexp would still match, but the trouble is in extracting the result with `match-string`.  The group in the start `*` will contain only the /last/ match (here, `rust-clippy`).

Reply:
Ah, yes, of course. Then a single `?` with a `+` *inside* the capturing group? The calling code would have to then further split the match (string-split on spaces)

Reply:
I think this is just fine as it is, though — we can deal with this problem later, if/when we need an extra addition